pkg/plugin/cmd/agent/check/check.go

```diff
@@ -0,0 +1,256 @@
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-operator/pkg/plugin/common"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const maxParallel = 10
+
+var (
+	checkExample = `
+  # check if the running Agents have detected check errors
+  %[1]s check
+`
+)
+
+// options provides information required by agent check command
+type options struct {
+	genericclioptions.IOStreams
+	configFlags   *genericclioptions.ConfigFlags
+	args          []string
+	clientset     *kubernetes.Clientset
+	restConfig    *restclient.Config
+	userNamespace string
+}
+
+// newOptions provides an instance of options with default values
+func newOptions(streams genericclioptions.IOStreams) *options {
+	return &options{
+		configFlags: genericclioptions.NewConfigFlags(false),
+		IOStreams:   streams,
+	}
+}
+
+// New provides a cobra command wrapping options
+func New(streams genericclioptions.IOStreams) *cobra.Command {
+	o := newOptions(streams)
+	cmd := &cobra.Command{
+		Use:          "check [flags]",
+		Short:        "Find check errors",
+		Example:      fmt.Sprintf(checkExample, "kubectl datadog agent"),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.complete(c, args); err != nil {
+				return err
+			}
+			if err := o.validate(); err != nil {
+				return err
+			}
+			return o.run()
+		},
+	}
+
+	o.configFlags.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+// complete sets all information required for processing the command
+func (o *options) complete(cmd *cobra.Command, args []string) error {
+	o.args = args
+	var err error
+
+	clientConfig := o.configFlags.ToRawKubeConfigLoader()
+
+	o.restConfig, err = o.configFlags.ToRawKubeConfigLoader().ClientConfig()
+	if err != nil {
+		return fmt.Errorf("unable to instantiate restConfig: %v", err)
+	}
+
+	o.clientset, err = common.NewClientset(clientConfig)
+	if err != nil {
+		return fmt.Errorf("unable to instantiate clientset: %v", err)
+	}
+
+	o.userNamespace, _, err = clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	ns, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+
+	if ns != "" {
+		o.userNamespace = ns
+	}
+
+	return nil
+}
+
+// validate ensures that all required arguments and flag values are provided
+func (o *options) validate() error {
+	return nil
+}
+
+// run runs the check command
+func (o *options) run() error {
+	agentPods, err := o.getPods("agent")
+	if err != nil {
+		return fmt.Errorf("unable to get Agent pods: %v", err)
+	}
+	clcPods, err := o.getPods("cluster-checks-runner")
+	if err != nil {
+		return fmt.Errorf("unable to get Agent pods: %v", err)
+	}
+	fmt.Println(fmt.Sprintf("Found %d node Agents and %d Cluster-Check Runners", len(agentPods), len(clcPods)))
```

Comment:
actually it provides some functions that fallback to Stderr if not set if `outWriter` is not set - I'm not seeing the reason why we should call these functions since we want to print to Stdout, am I missing something here?

Reply:
One major interest is to be able to unit-test the command output.